### PR TITLE
chore: refactor funcs to receive resource.Quantity type

### LINKED ISSUE #426: Refactor resource allocation functions to receive resource.Quantity type instead.
It would be nice to refactor these function to receive resource.Quantity type instead. We can do it in a follow up PR.
functions like these: `SetCPU`, `SetMemory` and `AddVolume`

_Originally posted by @mojtaba-esk in https://github.com/celestiaorg/knuu/pull/409#discussion_r1627288338_

### PR DESCRIPTION
**Pull request by mojtaba-esk**
Closes #426 

## Summary by CodeRabbit

- **Refactor**
  - Updated memory and CPU limit handling to use `resource.MustParse` for better resource management.
  - Renamed variables for better clarity in volume size handling.
  - Changed method signatures to accept `resource.Quantity` for memory, CPU, and volume sizes.

- **Improvements**
  - Enhanced accuracy and consistency in setting resource limits and requests.
  - Improved readability and maintainability of resource-related code.

These changes ensure more reliable and precise handling of CPU, memory, and volume resources, leading to better performance and stability.

### CODE REVIEW

File: pkg/instance/executor.go [hunk 1]
```diff
@@ -37,11 +39,11 @@ func NewExecutor(ctx context.Context, sysDeps system.SystemDependencies) (*Execu
 		return nil, ErrSettingArgs.Wrap(err)
 	}
 
-	if err := i.SetMemory(memoryLimit, memoryLimit); err != nil {
+	if err := i.SetMemory(resource.MustParse(memoryLimit), resource.MustParse(memoryLimit)); err != nil {
```
Comment (MSevey) [inline pkg/instance/executor.go:42]: We should validate these at compile by moving `memoryLimit` and also `cpuLimit` from `const` to `var`

```go
var (
	memoryLimit = resource.MustParse("100M")
	cpuLimit    = resource.MustParse("100m")
)
```

File: pkg/instance/otel.go [hunk 2]
```diff
@@ -191,10 +192,10 @@ func (i *Instance) createOtelCollectorInstance(ctx context.Context) (*Instance,
 	if err := otelAgent.AddPortTCP(9090); err != nil {
 		return nil, ErrAddingOtelAgentPort.Wrap(err)
 	}
-	if err := otelAgent.SetCPU("100m"); err != nil {
+	if err := otelAgent.SetCPU(resource.MustParse("100m")); err != nil {
```
Comment (MSevey) [inline pkg/instance/otel.go:195]: use `cpuLimit`

File: pkg/instance/otel.go [hunk 3]
```diff
@@ -191,10 +192,10 @@ func (i *Instance) createOtelCollectorInstance(ctx context.Context) (*Instance,
...
 		return nil, ErrSettingOtelAgentCPU.Wrap(err)
 	}
-	if err := otelAgent.SetMemory("100Mi", "200Mi"); err != nil {
+	if err := otelAgent.SetMemory(resource.MustParse("100Mi"), resource.MustParse("200Mi")); err != nil {
```
Comment (MSevey) [inline pkg/instance/otel.go:198]: Why aren't we using `memoryLimit` here? 

We should either use `memoryLimit` or define a new `memoryLimitOtel` with a comment explaining why it is different.

File: pkg/instance/tshark.go [hunk 4]
```diff
@@ -40,13 +42,21 @@ func (i *Instance) createTsharkCollectorInstance(ctx context.Context) (*Instance
 	if err := tsc.Commit(); err != nil {
 		return nil, err
 	}
-	if err := tsc.SetCPU(tsharkCollectorCPU); err != nil {
+	if err := tsc.SetCPU(resource.MustParse(tsharkCollectorCPU)); err != nil {
```
Comment (MSevey) [inline pkg/instance/tshark.go:45]: same comment about moving `tsharkCollectorCPU` to a var so we can validate at compile.

File: pkg/instance/tshark.go [hunk 5]
```diff
@@ -40,13 +42,21 @@ func (i *Instance) createTsharkCollectorInstance(ctx context.Context) (*Instance
...
+		resource.MustParse(tsharkCollectorMemory),
+		resource.MustParse(tsharkCollectorMemory),
+	)
+	if err != nil {
 		return nil, err
 	}
-	if err := tsc.AddVolume(tsharkCollectorVolumePath, i.tsharkCollectorConfig.VolumeSize); err != nil {
+	err = tsc.AddVolume(
+		tsharkCollectorVolumePath,
+		resource.MustParse(i.tsharkCollectorConfig.VolumeSize),
```
Comment (MSevey) [inline pkg/instance/tshark.go:57]: similar concept to validating at compile, we should push the validation upstream as far as possible. So we should make `tsharkCollectorConfig.VolumeSize` a `resource.Quantity`.

File: pkg/minio/minio.go [hunk 6]
```diff
@@ -128,7 +128,12 @@ func (m *Minio) createOrUpdateDeployment(ctx context.Context) error {
 	if err != nil {
 		if errors.IsNotFound(err) {
 			// Deployment does not exist, create it
-			if err := m.createPVC(ctx, VolumeClaimName, PVCStorageSize, metav1.CreateOptions{}); err != nil {
+			err := m.createPVC(ctx,
+				VolumeClaimName,
+				resource.MustParse(PVCStorageSize),
```
Comment (MSevey) [inline pkg/minio/minio.go:133]: Same, `PVCStorageSize` can be a variable.

**Review by MSevey — CHANGES_REQUESTED** [review]
nice clean up, we can do a little more :-)

File: pkg/instance/otel.go (hunk 3 above)
Comment (mojtaba-esk) [reply pkg/instance/otel.go:198]: @smuu do you know why is it different from executor?

File: pkg/instance/tshark.go (hunk 5 above)
Comment (mojtaba-esk) [reply pkg/instance/tshark.go:59]: oh, nice catch. How could this be swiped out of my hand!

**Review by MSevey — APPROVED** [review]
👏🏻

File: pkg/instance/executor.go [hunk 7]
```diff
@@ -11,8 +13,11 @@ const (
 	executorName         = "executor"
 	sleepCommand         = "sleep"
 	infinityArg          = "infinity"
-	memoryLimit          = "100M"
-	cpuLimit             = "100m"
+)
+
+var (
+	executorMemoryLimit = resource.MustParse("100M")
```
[the remark this answers is not in the document]
Comment (mojtaba-esk) [reply pkg/instance/executor.go:19]: I have no idea about these notations :D

**Pull request merged**